### PR TITLE
[PPML] Add FL Dockerfile with graphene

### PR DESCRIPTION
**Pull request by piaolaidelangman**
* Add Dockerfile.
* Add build-image.sh
* Add run-server-and-client.sh
* Change scala/ppml/pom.xml, add new dependency and relocation com.google.protobuf.

### CODE REVIEW

File: scala/ppml/demo/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    rm bigdl*.zip
+
+WORKDIR /opt
+ADD bigdl-ppml-spark_3.1.2-0.14.0-SNAPSHOT-jar-with-dependencies.jar /opt/bigdl-0.14.0-SNAPSHOT/jars
+ADD runFlserver.sh /opt
+ADD runHflClient1.sh /opt
+ADD runHflClient2.sh /opt
+ADD runVflClient1.sh /opt
+ADD runVflClient2.sh /opt
+ADD ./data /opt/data
```
Comment (qiyuangong) [inline scala/ppml/demo/Dockerfile:64]: Need to chmod +x for scripts

Comment (piaolaidelangman) [reply scala/ppml/demo/Dockerfile:64]: Updated in this [commit](https://github.com/intel-analytics/BigDL/pull/3843/commits/0ab3f5c8adb3d9aca2c3e81278595f3a4d05be55).

And I add graphene-java-makefile, manifest and start-with-sgx scripts in this [commit](https://github.com/intel-analytics/BigDL/pull/3843/commits/011983f1ae8645d9d2630d3b1619d50d30477d2f).

**Comment by glorysdj** [comment]
I suggest to from ppml-graphene as parent image, add vfl parts, to avoid duplicated code and files.

File: scala/ppml/demo/runHflClient1.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+SGX=1 graphene-sgx bash -c "/opt/jdk8/bin/java -cp /opt/bigdl-0.14.0-SNAPSHOT/jars/bigdl-ppml-spark_3.1.2-0.14.0-SNAPSHOT-jar-with-dependencies.jar \
+    com.intel.analytics.bigdl.ppml.example.HflLogisticRegression -d /opt/data/diabetes-hfl-1.csv"
```
Comment (qiyuangong) [inline scala/ppml/demo/runHflClient1.sh:3]: Add  empty line in each file

File: scala/ppml/demo/deploy-local-spark-sgx.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+export ENCLAVE_KEY_PATH=YOUR_LOCAL_ENCLAVE_KEY_PATH
+export DATA_PATH=YOUR_LOCAL_DATA_PATH
+export KEYS_PATH=YOUR_LOCAL_KEYS_PATH
+export LOCAL_IP=YOUR_LOCAL_IP
+export DOCKER_IMAGE=intelanalytics/bigdl-ppml-trusted-big-data-ml-python-graphene:0.14.0-SNAPSHOT
```
Comment (qiyuangong) [inline scala/ppml/demo/deploy-local-spark-sgx.sh:7]: Image name should be the same

File: scala/ppml/demo/runHflClient1.sh (hunk 2 above)
Comment (piaolaidelangman) [reply scala/ppml/demo/runHflClient1.sh]: Updated.

File: scala/ppml/demo/deploy-local-spark-sgx.sh (hunk 3 above)
Comment (piaolaidelangman) [reply scala/ppml/demo/deploy-local-spark-sgx.sh]: Rename to `intelanalytics/bigdl-ppml-trusted-big-data-fl-scala-graphene:flDemo`. Can I use this name?

**Comment by piaolaidelangman** [comment]
[Jenkins](http://10.112.231.51:18888/view/ZOO-PR/job/BigDL-PR-PPML-UTs-Validation/8/console) passed.

**Review by qiyuangong — APPROVED** [review]
LGTM

**Review by Litchilitchy — APPROVED** [review]

**Pull request merged**